src/shared/file_op.c

```diff
@@ -2024,24 +2024,30 @@ int TempFile(File *file, const char *source, int copy) {
         return -1;
     }
 
+    if(fp_src = fopen(source,"r"), !fp_src){
+        close(fd);
+        unlink(template);
+        return -1;
+    }
+
 #ifndef WIN32
     struct stat buf;
 
-    if (stat(source, &buf) == 0) {
-        if (fchmod(fd, buf.st_mode) < 0) {
-            close(fd);
-            unlink(template);
-            return -1;
-        }
-    } else {
+    if (fstat(fp_src->_fileno, &buf) != 0) {
         mdebug1(FSTAT_ERROR, source, errno, strerror(errno));
+        /*unlink(template);
```

Comment:
Don't we really want these commented lines? This would make `fchmod()` read an unitialized `buf` value.

Reply:
Uncommented.